src/Sylius/Bundle/CoreBundle/spec/EventListener/CustomerDefaultAddressListenerSpec.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Bundle\CoreBundle\EventListener;
+
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Sylius\Bundle\ResourceBundle\Event\ResourceControllerEvent;
+use Sylius\Component\Core\Model\AddressInterface;
+use Sylius\Component\Core\Model\CustomerInterface;
+
+final class CustomerDefaultAddressListenerSpec extends ObjectBehavior
+{
+    function let(): void
+    {
+        $this->beConstructedWith();
+    }
+
+    function it_add_address_as_default_on_pre_create_resource_controller_event(
+        ResourceControllerEvent $event,
+        AddressInterface $address,
+        CustomerInterface $customer
+    ): void {
+        $event->getSubject()->willReturn($address);
+        $address->getCustomer()->willReturn($customer);
+
+        $address->getId()->shouldBeCalled();
+        $address->getCustomer()->shouldBeCalled();
+
+        $customer->getDefaultAddress()->willReturn(null);
+        $customer->getDefaultAddress()->shouldBeCalled();
+
+        $customer->setDefaultAddress($address)->shouldBeCalled();
+
+        $this->preCreate($event);
+    }
+
+    function it_does_set_address_as_default_if_customer_does_not_had_one_on_pre_create(
```

Comment:
I believe it should be `it_does_not_set_address_as_default_if_customer_alredy_have_default_address` :)

Reply:
We should also check two more scenarios:
1. it does nothing if an address does not have ID
2. it does nothing if an address does not have a customer assigned